zero/src/block_interval.rs

```diff
@@ -1,16 +1,40 @@
 use std::pin::Pin;
 use std::sync::Arc;
+use std::{future::Future, ops::Range};
 
-use alloy::primitives::B256;
-use alloy::rpc::types::eth::BlockId;
-use alloy::{hex, providers::Provider, transports::Transport};
+use alloy::providers::Provider;
+use alloy::rpc::types::BlockTransactionsKind;
+use alloy::rpc::types::{eth::BlockId, Block};
 use anyhow::{anyhow, Result};
 use async_stream::try_stream;
 use futures::Stream;
+#[cfg(test)]
+use mockall::automock;
 use tracing::info;
 
-use crate::parsing;
-use crate::provider::CachedProvider;
+#[cfg_attr(test, automock)]
+pub trait BlockIntervalProvider {
```

Comment:
@0xaatif note that `auto_impl(&, Box)` caused conflicting implementations to `impl<T: Provider> BlockIntervalProvider for T {` so that its why its not here. My understanding is that the impl block below would cover refs and Boxes. LMK if I'm mistaken